crowbar_framework/lib/crowbar/lock/base.rb

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2016, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  module Lock
+    class Base
+      attr_accessor :logger
+      attr_accessor :path
+      attr_accessor :name
+
+      attr_reader :path
+      attr_reader :name
+      attr_reader :file
+
+      def initialize(options = {})
+        @logger = options.fetch :logger, Rails.logger
+        @name = options.fetch :name, "default.lock"
+        @path = options.fetch :path, Rails.root.join("tmp", @name)
+        @locked = false
+        @file = nil
+      end
+
+      def locked?
```

Comment:
I don't know how to write `attr_reader` for boolean attributes that match the convention with a `?` suffix.